ckanext/ontario_theme/plugin.py

```diff
@@ -369,14 +369,13 @@ def get_group(group_id):
         data_dict={'id': group_id})
     return group_dict
 
-def get_recently_updated_datasets():
-    '''Helper to return 3 freshest datasets
+def get_keyword_count(keyword):
+    '''Helper to return the dataset count of the indicated group
     '''
-    recently_updated_datasets = toolkit.get_action('package_search')(
-        data_dict={'rows': 3,
-                    'sort': 'current_as_of desc'})
-    return recently_updated_datasets['results']
-
+    tags = 'tags:"{}"'.format(keyword)
+    keyword_count = toolkit.get_action('package_search')(
```

Comment:
I see an issue here: by counting only the number of english keywords, this could cause a mis-match with the counts on the Dataset page when in French, in the case where an English keyword was entered but the French keyword was left blank. To get around this, try to find how keywords are counted in the dataset page and return that in the `get_keyword_count()`. Below are screenshots of the Dataset page in English and French showing 2 counts for Health in English and 1 count in French. On the homepage, 2 counts will be shown in both English and French, but then clicking on the Health icon only shows 1 dataset when navigating in French.

<img width="305" alt="Screen Shot 2022-12-15 at 10 43 26 AM" src="https://user-images.githubusercontent.com/1254764/207904221-e10f692e-fe64-4037-beb3-502554f3ff49.png">
<img width="350" alt="Screen Shot 2022-12-15 at 10 44 01 AM" src="https://user-images.githubusercontent.com/1254764/207904382-ed0fb50d-ce9d-40a0-b629-8756703846de.png">

Reply:
> ontario_theme/templates/internal/snippets/facet_list.html generates the datasets of each keyword differently, but I couldn't understand how to use it this way.
It's in this location if you want to take a look at it.

Reply:
@Alexandra-Green I couldn't find how to define `items` in that snippet in our html file either. Here's something else that works that you could try:

- in [ontario_theme_categories.html](https://github.com/ongov/ckanext-ontario_theme/blob/ckan_2.9_upgrade_topics_section/ckanext/ontario_theme/templates/internal/home/snippets/ontario_theme_categories.html#L44), set the language before the `for` loop: `{% set language = h.lang() %}`
- in [ontario_theme_categories.html L53](https://github.com/ongov/ckanext-ontario_theme/blob/ckan_2.9_upgrade_topics_section/ckanext/ontario_theme/templates/internal/home/snippets/ontario_theme_categories.html#L53), instead of passing in the english keyword to your function, pass the keyword in the language that the application is currently in, and also pass the language:

```
({{ h.ontario_theme_get_keyword_count(_(topics[i]['label']), language) }})
```

- modify [get_keyword_count](https://github.com/ongov/ckanext-ontario_theme/blob/ckan_2.9_upgrade_topics_section/ckanext/ontario_theme/plugin.py#L372) to count the keywords as per the language passed: 
```
def get_keyword_count(keyword_lang, language):
    '''Helper to return the dataset count of the indicated group by language
    '''
    facet = "keywords_{}".format(language)
    package_keywords = toolkit.get_action('package_search')(
    data_dict={'facet.field': [facet],
        'rows': 0})
    active_keywords = package_keywords['facets'][facet]
    if keyword_lang in active_keywords:
        keyword_count_by_lang = active_keywords[keyword_lang]
    else:
        keyword_count_by_lang = 0

    return keyword_count_by_lang
```